mooncake-store/src/master_service.cpp

```diff
@@ -463,13 +468,72 @@ void MasterService::GCThreadFunc() {
             VLOG(1) << "key=" << task->key << ", action=gc_removing_key";
             ErrorCode result = Remove(task->key);
             if (result != ErrorCode::OK &&
-                result != ErrorCode::OBJECT_NOT_FOUND) {
+                result != ErrorCode::OBJECT_NOT_FOUND &&
+                result != ErrorCode::OBJECT_HAS_LEASE) {
                 LOG(WARNING)
                     << "key=" << task->key
                     << ", error=gc_remove_failed, error_code=" << result;
             }
+            if (result == ErrorCode::OK) {
+                gc_count++;
+            }
             delete task;
         }
+        if (gc_count > 0) {
+            MasterMetricManager::instance().dec_key_count(gc_count);
+        }
+
+        if (need_eviction_) {
+            auto now = std::chrono::steady_clock::now();
+            long evicted_count = 0;
+            long object_count = 0;
+            uint64_t total_freed_size = 0;
+            for (auto& shard : metadata_shards_) {
```

Comment:
Should we be keeping track of the shards we've already processed, maybe by maintaining a `shard_idx`?  My concern is that without that, we might be consistently evicting objects from the earlier shards.  This could potentially lead to a lower hit ratio overall.  What do you think?

Reply:
The current code logic is that once eviction is needed, we iterate through all the shards, and for each shard we evict about 10% objects. The 10% is counted individually for each shard.